src/compiler/parser.ts

```diff
@@ -503,7 +503,13 @@ namespace ts {
 
     export function createSourceFile(fileName: string, sourceText: string, languageVersion: ScriptTarget, setParentNodes = false, scriptKind?: ScriptKind): SourceFile {
         performance.mark("beforeParse");
-        const result = Parser.parseSourceFile(fileName, sourceText, languageVersion, /*syntaxCursor*/ undefined, setParentNodes, scriptKind);
+        let result: SourceFile;
+        if (languageVersion === ScriptTarget.JSON) {

```

Comment:
The relevant incoming call stack here is via `host.getSourceFile`, which only accepts `ScriptTarget` rather than `ScriptKind`

Reply:
I think you want to handle it like in https://github.com/Microsoft/TypeScript/pull/22167/files#diff-ead24f0f0f59c0ea9c1c511052e8884bL671

Reply:
@sheetalkamat that seems like the wrong place. Parser exposes `parseSourceFile` and `parseJsonText` separately, if someone wants to get back a JSON file they can call `parseJsonText` instead instead of having `parseSourceFile` reroute it